{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true,
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/MSGanbold/YJ.Chen_Gliocidin/blob/main/YJ_Chen_et_al_Gliocidin_paper.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##YJ's CRISPR with MAGeCK"
      ],
      "metadata": {
        "id": "2FDwj-p0pG1J"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "conda create --name mageck-vispr mageck mageck-vispr python=3.7\n",
        "\n",
        "https://bitbucket.org/liulab/mageck-vispr/src/master/\n",
        "\n",
        "\n",
        "#Docker\n",
        "$ docker pull davidliwei/mageck-vispr\n",
        "$ docker run -d -p 80:80 docker/getting-started\n",
        "$ docker pull davidliwei/mageck-vispr\n",
        "$ docker run -it -d -v ~/Docker:/home/data -p 5000:5000 --cpus=\"6\" --privileged=true --name mageck-vi\n",
        "$ source activate mageck-vispr\n",
        "\n",
        "\n",
        "############################### TO RUN (prior: edit parameters in .yaml locally, make sure all the folder shown r there)\n",
        "$cd /home/data/\n",
        "$ls\n",
        "$cd workflow_mle_2/\n",
        "$vispr server --host 0.0.0.0 results/*.vispr.yaml\n",
        "\n",
        "$mageck-vispr init workflow_mle_2 --reads Fastq_Combined/*.fastq*\n",
        "($mageck-vispr init path/to/my/workflow --reads path/to/sample1.fastq path/to/sample2.fastq ...)\n",
        "\n",
        "$cd workflow_mle_2\n",
        "$snakemake -n\n",
        "$snakemake --cores 8\n",
        "\n",
        "$vispr server results/*.vispr.yaml\n",
        "\n",
        "docker stop mageck-vispr\n",
        "\n",
        "#collapsing duplicate sg genes to their unique mean\n",
        "aggregated_posit_negat_hits_df <- aggregate(posit_negat_hits_df[-1], posit_negat_hits_df[1], mean)\n",
        "write.csv(aggregated_posit_negat_hits_df, \"/Users/mungunsarnaiganbold/Desktop/YJChen/aggregated_posit_negat_hits_df.csv\") #629 genes\n",
        "dim(aggregated_posit_negat_hits_df)\n",
        "\n",
        "plot(x,y,xlab=\"PC1\",ylab=\"PC2\", col = ifelse(x < 0,'red','green'), pch = 19 )\n",
        "\n",
        "\n",
        "\n",
        "plot(x=aggregated_posit_negat_hits_df$DMSO, y=aggregated_posit_negat_hits_df$IC50,\n",
        "     xlab=\"DMSO\",ylab=\"IC50\", col = ifelse(x > 0,'red','blue'), pch = 19 )\n",
        "range(aggregated_posit_negat_hits_df$DMSO)\n",
        "range(aggregated_posit_negat_hits_df$IC50)\n",
        "\n",
        "subset<- cbind(aggregated_posit_negat_hits_df$DMSO.R1, aggregated_posit_negat_hits_df$DMSO.R2)\n",
        "aggregated_posit_negat_hits_df$DMSO <- rowMeans( as.matrix(subset) )\n",
        "\n",
        "subset2<- cbind(aggregated_posit_negat_hits_df$IC50.R1, aggregated_posit_negat_hits_df$IC50.R2)\n",
        "aggregated_posit_negat_hits_df$IC50 <- rowMeans( as.matrix(subset2) )\n",
        "\n",
        "\n",
        "subset3<- cbind(aggregated_posit_negat_hits_df$IC80.R1, aggregated_posit_negat_hits_df$IC80.R2)\n",
        "aggregated_posit_negat_hits_df$IC80 <- rowMeans( as.matrix(subset3) )\n",
        "\n",
        "aggregated_posit_negat_hits_df$log2DMSO<- log2(aggregated_posit_negat_hits_df$DMSO)\n",
        "aggregated_posit_negat_hits_df$log2IC50<- log2(aggregated_posit_negat_hits_df$IC50)\n",
        "aggregated_posit_negat_hits_df$log2IC80<- log2(aggregated_posit_negat_hits_df$IC80)\n",
        "\n",
        "dim(aggregated_posit_negat_hits_df) #629  15\n",
        "\n",
        "\n",
        "plot(x=aggregated_posit_negat_hits_df$log2DMSO, y=aggregated_posit_negat_hits_df$log2IC50,\n",
        "     xlab=\"log2(DMSO)\",ylab=\"log2(IC50)\", col = \"grey\", pch = 19 )\n",
        "\n",
        "plot(x=aggregated_posit_negat_hits_df$log2DMSO, y=aggregated_posit_negat_hits_df$log2IC80,\n",
        "     xlab=\"log2(DMSO)\",ylab=\"log2(IC80)\", col = \"darkgrey\", pch = 19 )\n",
        "\n",
        "\n",
        "ggscatter(aggregated_posit_negat_hits_df, x = \"log2DMSO\", y = \"log2IC50\",\n",
        "          add = \"reg.line\", conf.int = FALSE,\n",
        "          cor.coef = TRUE, cor.method = \"spearman\",\n",
        "          xlab = \"log2(DMSO)\", ylab = \"log2(IC50)\",\n",
        "          color = \"blue\")\n",
        "\n",
        "ggscatter(aggregated_posit_negat_hits_df, x = \"log2DMSO\", y = \"log2IC80\",\n",
        "          add = \"reg.line\", conf.int = FALSE,\n",
        "          cor.coef = TRUE, cor.method = \"spearman\",\n",
        "          xlab = \"log2(DMSO)\", ylab = \"log2(IC80)\",\n",
        "          color = \"navy\")\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "#COUNTS ALL, AVERAGE REPLICATES, LOG2\n",
        " data12<- cbind(counts$IC80.R1, counts$IC80.R2)\n",
        " counts$IC80 <- rowMeans( as.matrix(data12) )\n",
        "\n",
        " data13<- cbind(counts$IC50.R1, counts$IC50.R2)\n",
        " counts$IC50 <- rowMeans( as.matrix(data13) )\n",
        "\n",
        " data15<- cbind(counts$DMSO.R1, counts$DMSO.R2)\n",
        " counts$DMSO <- rowMeans( as.matrix(data15) )\n",
        "\n",
        " counts$log2IC50 <- log2(counts$IC50)\n",
        " counts$log2IC80 <- log2(counts$IC80)\n",
        " counts$log2DMSO <- log2(counts$DMSO)\n",
        "\n",
        "\n",
        "ggscatter(counts, x = \"log2DMSO\", y = \"log2IC80\",\n",
        "           add = \"reg.line\", conf.int = FALSE,\n",
        "           cor.coef = TRUE, cor.method = \"spearman\",\n",
        "           xlab = \"gRNA abundance in DMSO treated cells (log2)\",\n",
        "          ylab = \"gRNA abundance in IC80 treated cells (log2)\",\n",
        "           color = \"darkgrey\")\n",
        "\n",
        "ggscatter(counts, x = \"log2DMSO\", y = \"log2IC50\",\n",
        "          add = \"reg.line\", conf.int = FALSE,\n",
        "          cor.coef = TRUE, cor.method = \"spearman\",\n",
        "          xlab = \"gRNA abundance in DMSO treated cells (log2)\",\n",
        "          ylab = \"gRNA abundance in IC50 treated cells (log2)\",\n",
        "          color = \"grey\")\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "#HEATMAP\n",
        "heatmap_df <- counts[which(counts$Gene %in% c(\"Gart\", \"Ppat\", \"Atic\", \"Tsc2\", \"Nprl3\", \"Nprl2\")), ]\n",
        "heatmap_df <- heatmap_df[,c(2:10)]\n",
        "colnames(heatmap_df)\n",
        "dim(heatmap_df) #multiple sg RNA\n",
        "\n",
        "#collapsing duplicate sg genes to their unique mean\n",
        "aggregated_heatmap_df <- aggregate(heatmap_df[-1], heatmap_df[1], mean)\n",
        "dim(aggregated_heatmap_df) #6 9\n",
        "rownames(รง) <- aggregated_heatmap_df$Gene\n",
        "aggregated_heatmap_df <- aggregated_heatmap_df[,-1]\n",
        "\n",
        "\n",
        "library(gplots)\n",
        "\n",
        "heatmap_df<- aggregated_heatmap_df[,-c(1,2)]\n",
        "\n",
        "heatmap.2(as.matrix(heatmap_df), scale = \"row\",\n",
        "          col = bluered(100),\n",
        "          #col = colorRampPalette( rev(brewer.pal(9, \"RdBu\")) ),\n",
        "          trace = \"none\",\n",
        "          density.info = \"density\",\n",
        "          dendrogram = \"none\")\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "library(ggplot2)\n",
        "#Boxplots\n",
        "\n",
        "\n",
        "\n",
        "data_t <- as.data.frame( t(data) )\n",
        "data_t <- data_t[,c(6,3,4, 2, 5,1)]\n",
        "#three boxes/cond next to each other\n",
        "\n",
        "par(mfrow=c(1,3))\n",
        "boxplot( data_t[3,1],\n",
        "         data_t[3,2],\n",
        "         data_t[3,3],\n",
        "         data_t[3,4],\n",
        "         data_t[3,5],\n",
        "         data_t[3,6],\n",
        "         main = \"DMSO\",\n",
        "         #xlab = c(\"Tsc2\",\"Nprl2\",\"Nprl3\",\"Gart\",\"Ppat\",\"Atic\"),\n",
        "         ylab = \"log2 normalized counts\",\n",
        "         at = c(1,2,3,4,5,6),\n",
        "         col = c(\"blue\", \"blue\", \"blue\", \"red\", \"red\", \"red\"),\n",
        "         border = \"black\",\n",
        "         horizontal = F,\n",
        "         notch = TRUE)\n",
        "boxplot( data_t[2,1],\n",
        "         data_t[2,2],\n",
        "         data_t[2,3],\n",
        "         data_t[2,4],\n",
        "         data_t[2,5],\n",
        "         data_t[2,6],\n",
        "         main = \"IC50\",\n",
        "         #xlab = c(\"Tsc2\",\"Nprl2\",\"Nprl3\",\"Gart\",\"Ppat\",\"Atic\"),\n",
        "         ylab = \"log2 normalized counts\",\n",
        "         at = c(1,2,3,4,5,6),\n",
        "         col = c(\"blue\", \"blue\", \"blue\", \"red\", \"red\", \"red\"),\n",
        "         border = \"black\",\n",
        "         horizontal = F,\n",
        "         notch = TRUE)\n",
        "boxplot( data_t[1,1],\n",
        "         data_t[1,2],\n",
        "         data_t[1,3],\n",
        "         data_t[1,4],\n",
        "         data_t[1,5],\n",
        "         data_t[1,6],\n",
        "         main = \"IC80\",\n",
        "         #xlab = c(\"Tsc2\",\"Nprl2\",\"Nprl3\",\"Gart\",\"Ppat\",\"Atic\"),\n",
        "         ylab = \"log2 normalized counts\",\n",
        "         at = c(1,2,3,4,5,6),\n",
        "         col = c(\"blue\", \"blue\", \"blue\", \"red\", \"red\", \"red\"),\n",
        "         border = \"black\",\n",
        "         horizontal = F,\n",
        "         notch = TRUE)\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "#BETTA SCORES CURVES FOR IC50 vs DMSO\n",
        "mle_dmso_ic50 <- read.csv(\"~/mle_dmso_ic50.gene_summary.txt\", header=T, sep=\"\\t\")\n",
        "dim(dmso_ic50_p)\n",
        "betta_curve_input_ic50_ <- mle_dmso_ic50[order(-mle_dmso_ic50$IC50.beta), ]\n",
        "write.csv(betta_curve_input_ic50_, \"~betta_curve_input_ic50_.csv\")\n",
        "plot(betta_curve_input_ic50_$IC50.beta, main =\"DMSO vs. IC50\", xlab=\"Gene rank\", ylab=\"Genes Fitness (beta-score)\"\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "58XdpBWRpLYs"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "# YJ's single cell rna data"
      ],
      "metadata": {
        "id": "NWR-8V5fHr3A"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Alignment GEX with Cellranger"
      ],
      "metadata": {
        "id": "tWe5YFwEjW2g"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#HPC\n",
        "\n",
        "#!/bin/bash\n",
        "\n",
        "module load cellranger/7.1.0 #6.1.1\n",
        "\n",
        "\n",
        "cellranger count --id=Veh_out --fastqs= ~/FASTQ/Sample_Veh_IGO_14841_1 --sample=Veh_IGO_14841_1 --transcriptome=~/Cell_Ranger/refdata-cellranger-GRCh38-3.0.0\n",
        "\n",
        "cellranger count --id=Tmz_out --fastqs=~/FASTQ/Sample_TMZ_IGO_14841_3 --sample=TMZ_IGO_14841_3 --transcriptome=~/Cell_Ranger/refdata-cellranger-GRCh38-3.0.0\n",
        "\n",
        "cellranger count --id=Cpd_out --fastqs=~/FASTQ/Sample_Cpd_IGO_14841_2 --sample=Cpd_IGO_14841_2 --transcriptome=~/Cell_Ranger/refdata-cellranger-GRCh38-3.0.0\n",
        "\n",
        "cellranger count --id=Combo_out --fastqs=~/FASTQ/Sample_Combo_IGO_14841_4 --sample=Combo_IGO_14841_4 --transcriptome=~/Cell_Ranger/refdata-cellranger-GRCh38-3.0.0\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "RIAJhSEojhBH"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Combo bulk SC QC"
      ],
      "metadata": {
        "id": "iaiP33sN_Cir"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "library(Seurat)\n",
        "library(AUCell)\n",
        "\n",
        "#######################################################\n",
        "################  Combo ###################################\n",
        "########################################################\n",
        "counts <- Read10X(data.dir = \"/Volumes/paradalab/ParadaLab_Data/YJC/2023_RNAsc_Veh.Cpd.Tmz.Combo_IGO/Sarnai_CellRanger_counts.pipeline_w.o.multiplx/Combo_out/outs/filtered_feature_bc_matrix/\" )\n",
        "dim(counts) # 33538 20968\n",
        "Combo_obj <- CreateSeuratObject(counts = counts, project = \"Combo\")\n",
        "Combo_obj[[\"percent.mt\"]] <- PercentageFeatureSet(Combo_obj, pattern = \"^MT-\")\n",
        "\n",
        "setwd(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Combo/\")\n",
        "library(ggplot2)\n",
        "png(\"VLN_raw.png\")\n",
        "VlnPlot(Combo_obj, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "dim(Combo_obj[[]])\n",
        "\n",
        "\n",
        "\n",
        "######\n",
        "### subset1 lyzing cells by mt%\n",
        "png(\"hist_MT_raw.png\")\n",
        "hist(log10(Combo_obj$percent.mt))\n",
        "dev.off()\n",
        "\n",
        "png(\"raw_scatter_MT_UMI.png\")\n",
        "plot(Combo_obj$percent.mt, Combo_obj$nCount_RNA)\n",
        "dev.off()\n",
        "\n",
        "# decide\n",
        "subs1 <- subset(Combo_obj, subset = percent.mt < 10)\n",
        "dim(subs1)#\n",
        "png(\"VLN_subs1.png\")\n",
        "VlnPlot(subs1, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "### subs2 doublet/mult\n",
        "\n",
        "png(\"subs1_scatter_UMI_nF.png\")\n",
        "plot(subs1$nCount_RNA, subs1$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "png(\"hist_nC_subs1.png\")\n",
        "hist(log10(subs1$nCount_RNA))\n",
        "dev.off()\n",
        "\n",
        "subs2 <- subset(subs1, subset = nCount_RNA < 50000 & nFeature_RNA > 2000)\n",
        "dim(subs2)\n",
        "\n",
        "png(\"subs1_VLN.png\")\n",
        "VlnPlot(subs1, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "#### final result subset2\n",
        "\n",
        "#decide\n",
        "subs2 <- subset(subs1, subset = nCount_RNA < 10000)\n",
        "dim(subs2)\n",
        "png(\"VLN_subs2.png\")\n",
        "VlnPlot(subs2, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "png(\"subs2_scatter_UMI_nF.png\")\n",
        "plot(subs2$nCount_RNA, subs2$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "png(\"subs2_scatter_UMI_mt.png\")\n",
        "plot(subs2$nCount_RNA, subs2$percent.mt)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "saveRDS(subs2, \"Combo_obj_S1_subs2.rds\")\n",
        "\n",
        "Combo_obj_S1_subs2 <- subs2\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "W5foSgFc_QvK"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Tmz bulk SC QC"
      ],
      "metadata": {
        "id": "jafnpVLL_Sif"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "### YJC bulk SC RNA by four conditions #####\n",
        "\n",
        "library(Seurat)\n",
        "library(AUCell)\n",
        "\n",
        "#######################################################\n",
        "################  Tmz ###################################\n",
        "########################################################\n",
        "counts <- Read10X(data.dir = \"/Volumes/paradalab/ParadaLab_Processed.Data_to.be.sorted/YJC/2023_RNAsc_Veh.Cpd.Tmz.Combo_IGO/Sarnai_CellRanger_counts.pipeline_w.o.multiplx/Tmz_out/outs/filtered_feature_bc_matrix/\" )\n",
        "dim(counts)\n",
        "Tmz_obj <- CreateSeuratObject(counts = counts, project = \"Tmz\")\n",
        "Tmz_obj[[\"percent.mt\"]] <- PercentageFeatureSet(Tmz_obj, pattern = \"^MT-\")\n",
        "\n",
        "setwd(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Tmz/\")\n",
        "library(ggplot2)\n",
        "png(\"VLN_raw.png\")\n",
        "VlnPlot(Tmz_obj, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "dim(Tmz_obj[[]])\n",
        "\n",
        "\n",
        "\n",
        "######\n",
        "### subset1 lyzing cells by mt%\n",
        "png(\"hist_MT_raw.png\")\n",
        "hist(log10(Tmz_obj$percent.mt))\n",
        "dev.off()\n",
        "\n",
        "png(\"raw_scatter_MT_UMI.png\")\n",
        "plot(Tmz_obj$percent.mt, Tmz_obj$nCount_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "# decide\n",
        "subs1 <- subset(Tmz_obj, subset = percent.mt < 16)\n",
        "dim(subs1)#\n",
        "png(\"VLN_subs1.png\")\n",
        "VlnPlot(subs1, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "### subs2 doublet/mult\n",
        "\n",
        "png(\"subs1_scatter_UMI_nF.png\")\n",
        "plot(subs1$nCount_RNA, subs1$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "png(\"hist_nC_subs1.png\")\n",
        "hist(log10(subs1$nCount_RNA))\n",
        "dev.off()\n",
        "\n",
        "subs2 <- subset(subs1, subset = nCount_RNA < 36000 & nCount_RNA > 2000)\n",
        "dim(subs2)\n",
        "\n",
        "png(\"subs2_VLN.png\")\n",
        "VlnPlot(subs2, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "png(\"subs2_scatter_UMI_nF.png\")\n",
        "plot(subs2$nCount_RNA, subs2$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "png(\"subs2_scatter_UMI_mt.png\")\n",
        "plot(subs2$nCount_RNA, subs2$percent.mt)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "saveRDS(subs2, \"Tmz_obj_S1_subs2.rds\")\n",
        "\n",
        "Tmz_obj_S1_subs2 <- subs2\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "vbiYGOSv_VHx"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Cpd bulk SC QC"
      ],
      "metadata": {
        "id": "R5MIpnmIOJIW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "### YJC bulk SC RNA by four conditions #####\n",
        "\n",
        "\n",
        "library(Seurat)\n",
        "library(AUCell)\n",
        "\n",
        "#######################################################\n",
        "################  Cpd ###################################\n",
        "########################################################\n",
        "counts <- Read10X(data.dir = \"/Volumes/paradalab/ParadaLab_Processed.Data_to.be.sorted/YJC/2023_RNAsc_Veh.Cpd.Tmz.Combo_IGO/Sarnai_CellRanger_counts.pipeline_w.o.multiplx/Cpd_out/outs/filtered_feature_bc_matrix/\" )\n",
        "dim(counts)\n",
        "Cpd_obj <- CreateSeuratObject(counts = counts, project = \"Cpd\")\n",
        "Cpd_obj[[\"percent.mt\"]] <- PercentageFeatureSet(Cpd_obj, pattern = \"^MT-\")\n",
        "\n",
        "setwd(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Cpd/\")\n",
        "library(ggplot2)\n",
        "png(\"VLN_raw.png\")\n",
        "VlnPlot(Cpd_obj, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "dim(Cpd_obj[[]])\n",
        "\n",
        "### subset1 lyzing cells by mt%\n",
        "png(\"hist_MT_raw.png\")\n",
        "hist(log10(Cpd_obj$percent.mt))\n",
        "dev.off()\n",
        "\n",
        "png(\"raw_scatter_MT_UMI.png\")\n",
        "plot(Cpd_obj$percent.mt, Cpd_obj$nCount_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "# decide\n",
        "subs1 <- subset(Cpd_obj, subset = percent.mt < 18)\n",
        "dim(subs1)#\n",
        "png(\"VLN_subs1.png\")\n",
        "VlnPlot(subs1, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "### subs2 doublet/mult\n",
        "\n",
        "png(\"subs1_scatter_UMI_nF.png\")\n",
        "plot(subs1$nCount_RNA, subs1$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "png(\"hist_nC_subs1.png\")\n",
        "hist(log10(subs1$nCount_RNA))\n",
        "dev.off()\n",
        "\n",
        "subs2 <- subset(subs1, subset = nCount_RNA < 35000 & nCount_RNA > 1200)\n",
        "dim(subs2)\n",
        "\n",
        "png(\"subs2_VLN.png\")\n",
        "VlnPlot(subs2, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "png(\"subs2_scatter_UMI_nF.png\")\n",
        "plot(subs2$nCount_RNA, subs2$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "png(\"subs2_scatter_UMI_mt.png\")\n",
        "plot(subs2$nCount_RNA, subs2$percent.mt)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "saveRDS(subs2, \"Cpd_obj_S1_subs2.rds\")\n",
        "\n",
        "Cpd_obj_S1_subs2 <- subs2\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "pcMOyGfCOMKN"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Veh bulk SC QC"
      ],
      "metadata": {
        "id": "cebQ8c4FVuAa"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "### YJC bulk SC RNA by four conditions #####\n",
        "\n",
        "\n",
        "library(Seurat)\n",
        "library(AUCell)\n",
        "\n",
        "#######################################################\n",
        "################  Veh ###################################\n",
        "########################################################\n",
        "counts <- Read10X(data.dir = \"/Volumes/paradalab/ParadaLab_Processed.Data_to.be.sorted/YJC/2023_RNAsc_Veh.Cpd.Tmz.Combo_IGO/Sarnai_CellRanger_counts.pipeline_w.o.multiplx/Veh_out/outs/filtered_feature_bc_matrix/\" )\n",
        "dim(counts)\n",
        "Veh_obj <- CreateSeuratObject(counts = counts, project = \"Veh\")\n",
        "Veh_obj[[\"percent.mt\"]] <- PercentageFeatureSet(Veh_obj, pattern = \"^MT-\")\n",
        "\n",
        "setwd(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Veh/\")\n",
        "library(ggplot2)\n",
        "png(\"VLN_raw.png\")\n",
        "VlnPlot(Veh_obj, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "dim(Veh_obj[[]])\n",
        "\n",
        "\n",
        "\n",
        "######\n",
        "### subset1 lyzing cells by mt%\n",
        "png(\"hist_MT_raw.png\")\n",
        "hist(log10(Veh_obj$percent.mt))\n",
        "dev.off()\n",
        "\n",
        "png(\"raw_scatter_MT_UMI.png\")\n",
        "plot(Veh_obj$percent.mt, Veh_obj$nCount_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "# decide\n",
        "subs1 <- subset(Veh_obj, subset = percent.mt < 17)\n",
        "dim(subs1)#\n",
        "png(\"VLN_subs1.png\")\n",
        "VlnPlot(subs1, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "### subs2 doublet/mult\n",
        "\n",
        "png(\"subs1_scatter_UMI_nF.png\")\n",
        "plot(subs1$nCount_RNA, subs1$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "png(\"hist_nC_subs1.png\")\n",
        "hist(log10(subs1$nCount_RNA))\n",
        "dev.off()\n",
        "dim(subs1)\n",
        "\n",
        "subs2 <- subset(subs1, subset = nCount_RNA < 36000 & nCount_RNA > 1200)\n",
        "dim(subs2)\n",
        "\n",
        "png(\"subs2_VLN.png\")\n",
        "VlnPlot(subs2, features=c(\"nFeature_RNA\", \"nCount_RNA\", \"percent.mt\"), ncol=3, pt.size=0) + stat_summary(fun=mean, geom=\"point\", size=12, colour=\"black\", shape=95)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "png(\"subs2_scatter_UMI_nF.png\")\n",
        "plot(subs2$nCount_RNA, subs2$nFeature_RNA)\n",
        "dev.off()\n",
        "\n",
        "png(\"subs2_scatter_UMI_mt.png\")\n",
        "plot(subs2$nCount_RNA, subs2$percent.mt)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "saveRDS(subs2, \"Veh_obj_S1_subs2.rds\")\n",
        "\n",
        "Veh_obj_S1_subs2 <- subs2\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "Kqk2eeG2VvIr"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## INTEGRATED"
      ],
      "metadata": {
        "id": "btC8DPczaw40"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "\n",
        "setwd(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/integrated/\")\n",
        "\n",
        "\n",
        "Veh <- readRDS(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Veh/Veh_obj_S1_subs2.rds\")\n",
        "Tmz <- readRDS(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Tmz/Tmz_obj_S1_subs2.rds\")\n",
        "Cpd <- readRDS(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Cpd/Cpd_obj_S1_subs2.rds\")\n",
        "Cmb <- readRDS(\"/Volumes/paradalab/Sarnai_WIP/Swathi_and_YuJung/Yu_Jung/YJC_sc.editors_Oct2023/Combo/Combo_obj_S1_subs2.rds\")\n",
        "\n",
        "\n",
        "list <- list(Veh, Tmz, Cpd, Cmb)\n",
        "\n",
        "length(list) #4\n",
        "\n",
        "list <- lapply(X = list, FUN = function(x) {\n",
        "  x <- NormalizeData(x)\n",
        "  x <- FindVariableFeatures(x, selection.method = \"vst\", nfeatures = 2000)\n",
        "})\n",
        "\n",
        "\n",
        "features <- SelectIntegrationFeatures(object.list = list)\n",
        "list <- lapply(X = list, FUN = function(x) {\n",
        "  x <- ScaleData(x, features = features, verbose = FALSE)\n",
        "  x <- RunPCA(x, features = features, verbose = FALSE)\n",
        "})\n",
        "\n",
        "anchors <- FindIntegrationAnchors(object.list = list, anchor.features = features, reduction = \"cca\")\n",
        "saveRDS(anchors, \"anchors.rds\")\n",
        "# An AnchorSet object containing 19706 anchors between 4 Seurat objects\n",
        "# This can be used as input to IntegrateData\n",
        "\n",
        "UMAP_int <- IntegrateData(anchorset = anchors)\n",
        "DefaultAssay(UMAP_int) <- \"integrated\"\n",
        "UMAP_int <- ScaleData(UMAP_int, verbose = FALSE)\n",
        "UMAP_int <- RunPCA(UMAP_int, npcs = 30, verbose = FALSE)\n",
        "ElbowPlot(UMAP_int)\n",
        "UMAP_int <- RunUMAP(UMAP_int, reduction = \"pca\", dims = 1:20)\n",
        "\n",
        "UMAP_int <- FindNeighbors(UMAP_int, reduction = \"pca\", dims = 1:20)\n",
        "\n",
        "UMAP_int <- FindClusters(UMAP_int, resolution = 0.5)#\n",
        "UMAP_int <- RunUMAP(UMAP_int, dims = 1:20)\n",
        "pdf(\"UMAP_INT_res0.5.pdf\")\n",
        "DimPlot(UMAP_int, reduction = \"umap\", label=T)\n",
        "dev.off()\n",
        "\n",
        "UMAP_int <- FindClusters(UMAP_int, resolution = 0.3)#\n",
        "UMAP_int <- RunUMAP(UMAP_int, dims = 1:20)\n",
        "pdf(\"UMAP_INT_res0.3.pdf\")\n",
        "DimPlot(UMAP_int, reduction = \"umap\", label=T)\n",
        "dev.off()\n",
        "\n",
        "UMAP_int <- FindClusters(UMAP_int, resolution = 0.2)#\n",
        "UMAP_int <- RunUMAP(UMAP_int, dims = 1:20)\n",
        "pdf(\"UMAP_INT_res0.2.pdf\")\n",
        "DimPlot(UMAP_int, reduction = \"umap\", label=T)\n",
        "dev.off()\n",
        "\n",
        "UMAP_int <- FindClusters(UMAP_int, resolution = 0.7)#\n",
        "UMAP_int <- RunUMAP(UMAP_int, dims = 1:20)\n",
        "pdf(\"UMAP_INT_res0.7.pdf\")\n",
        "DimPlot(UMAP_int, reduction = \"umap\", label=T)\n",
        "dev.off()\n",
        "\n",
        "saveRDS(UMAP_int, \"obj.rds\")\n",
        "\n",
        "\n",
        "pdf(\"UMAP0.5_split_byOrig.ident.pdf\", width=25, height=5)\n",
        "DimPlot(UMAP_int, reduction = \"umap\", split.by=\"orig.ident\", group.by=\"integrated_snn_res.0.5\")\n",
        "dev.off()\n",
        "\n",
        "pdf(\"UMAP0.3_split_byOrig.ident.pdf\", width=25, height=5)\n",
        "DimPlot(UMAP_int, reduction = \"umap\", split.by=\"orig.ident\", group.by=\"integrated_snn_res.0.3\")\n",
        "dev.off()\n",
        "\n",
        "pdf(\"UMAP0.2_split_byOrig.ident.pdf\", width=25, height=5)\n",
        "DimPlot(UMAP_int, reduction = \"umap\", split.by=\"orig.ident\", group.by=\"integrated_snn_res.0.2\")\n",
        "dev.off()\n",
        "\n",
        "pdf(\"UMAP0.7_split_byOrig.ident.pdf\", width=25, height=5)\n",
        "DimPlot(UMAP_int, reduction = \"umap\", split.by=\"orig.ident\", group.by=\"integrated_snn_res.0.7\")\n",
        "dev.off()\n"
      ],
      "metadata": {
        "id": "ITCBD57way0F"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#extra plots\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split_GS_0.23.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"GS_0.23\", split.by = \"orig.ident\", cols=c(\"lightgrey\", \"darkgreen\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split__MIX_0.3.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"MIX_0.3\", split.by = \"orig.ident\", cols=c(\"lightgrey\", \"grey17\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split__P.S_0.2.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"P.S_0.2\", , split.by = \"orig.ident\", cols=c(\"lightgrey\", \"red3\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split__P.M_0.23.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"P.M_0.23\", split.by = \"orig.ident\", cols=c(\"lightgrey\", \"coral1\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split__OL_0.17.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"OL_0.17\", split.by = \"orig.ident\", cols=c(\"lightgrey\", \"blue\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/FP.split__N_0.23.pdf\", width = 12, height=4)\n",
        "FeaturePlot(obj_w.o.dead.cells, \"N_0.23\", split.by = \"orig.ident\", cols=c(\"lightgrey\", \"purple\"))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "##############\n",
        "\n",
        "\n",
        "\n",
        "\n",
        "Idents(obj_w.o.dead.cells)\n",
        "\n",
        "cols5=c(\"darkgreen\", \"darkgrey\", \"red3\", \"coral1\", \"blue\")\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/UMAP.res0.3_split.pdf\", width = 12, height=4)\n",
        "DimPlot(obj_w.o.dead.cells, split.by = \"orig.ident\", cols=cols5, pt.size = 0.05)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "pdf(\"/Volumes/paradalab/Sarnai/Yu_Jung/FinalPlots/UMAP.res0.3_not.split.pdf\")\n",
        "DimPlot(obj_w.o.dead.cells, cols=cols5, pt.size = 0.05)\n",
        "dev.off()\n",
        "\n",
        "\n",
        "\n",
        "saveRDS(obj_w.o.dead.cells, \"obj_w.o.dead.cells.rds\")\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "kCQccT9PHsKk"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "##### checking the tally table #########\n",
        "\n",
        "yj<- readRDS(\"/Volumes/paradalab/Sarnai/Yu_Jung/int_obj2_w.o.5.7.10.rds\")\n",
        "\n",
        "\n",
        "Idents(yj) <- yj$integrated_snn_res.0.3\n",
        "Idents(yj)\n",
        "\n",
        "yj$GS_0.23 <- ifelse(yj$GS.GS_auc_score>0.23, 100, 0)\n",
        "yj$MIX_0.3 <- ifelse(yj$MIX.MIX_auc_score>0.3, 100, 0)\n",
        "yj$P.S_0.2 <- ifelse(yj$P.S.P.S_auc_score>0.2, 100, 0)\n",
        "yj$P.M_0.23 <- ifelse(yj$P.M.P.M_auc_score>0.23, 100, 0)\n",
        "yj$OL_0.17 <- ifelse(yj$OL.OL_auc_score>0.17, 100, 0)\n",
        "\n",
        "m1 <- yj[[]] [which(yj$GS_0.23==\"100\"), ]\n",
        "m2 <- yj[[]] [which(yj$MIX_0.3==\"100\"), ]\n",
        "m3 <- yj[[]] [which(yj$P.S_0.2 ==\"100\"), ]\n",
        "m4 <- yj[[]] [which(yj$P.M_0.23 ==\"100\"), ]\n",
        "m5 <- yj[[]] [which(yj$OL_0.17 ==\"100\"), ]\n",
        "\n",
        "summary(as.factor(m1$integrated_snn_res.0.3))\n",
        "summary(as.factor(m2$integrated_snn_res.0.3))\n",
        "summary(as.factor(m3$integrated_snn_res.0.3))\n",
        "summary(as.factor(m4$integrated_snn_res.0.3))\n",
        "summary(as.factor(m5$integrated_snn_res.0.3))\n",
        "\n",
        "summary(as.factor(yj$orig.ident))\n",
        "Combo   Cpd   Tmz   Veh\n",
        "17522 21075 21413 24637\n",
        "\n",
        "\n",
        "\n",
        "print(\"COPY\")\n",
        "summary(as.factor(m1[which(m1$GS_0.23==\"100\" & m1$integrated_snn_res.0.3==\"3\"), ]$orig.ident))\n",
        "summary(as.factor(m2[which(m2$MIX_0.3==\"100\" & m2$integrated_snn_res.0.3==\"2\" | m2$integrated_snn_res.0.3==\"6\"), ]$orig.ident))\n",
        "summary(as.factor(m3[which(m3$P.S_0.2==\"100\" & m3$integrated_snn_res.0.3==\"1\"), ]$orig.ident))\n",
        "summary(as.factor(m4[which(m4$P.M_0.23==\"100\" & m4$integrated_snn_res.0.3==\"4\"), ]$orig.ident))\n",
        "summary(as.factor(m5[which(m5$OL_0.17==\"100\" & m5$integrated_snn_res.0.3==\"0\" | m5$integrated_snn_res.0.3==\"8\" | m5$integrated_snn_res.0.3==\"9\"), ]$orig.ident))\n",
        "\n",
        "\n",
        "saveRDS(yj, \"yjc_w.o.dead.cells_Apr.2024.rds\" )"
      ],
      "metadata": {
        "id": "OI1LM2ulGg6o"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## NMNAT1 Expression by 4 conds"
      ],
      "metadata": {
        "id": "CwInPJnvBFwr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "summary(as.factor(UMAP_int$orig.ident))\n",
        "UMAP_int$conds\n",
        "\n",
        "UMAP_int<- AddModuleScore(UMAP_int, features=\"NMNAT1\", name=\"NMNAT1_ModuleScore\", seed=324)\n",
        "range(UMAP_int$NMNAT1_ModuleScore1)\n",
        "mean(UMAP_int$NMNAT1_ModuleScore1)\n",
        "\n",
        "png(\"~/Downloads/hist_log10_NMNAT1.Score.png\")\n",
        "hist(log10(UMAP_int$NMNAT1_ModuleScore1 +1) )\n",
        "dev.off()\n",
        "\n",
        "png(\"~/Downloads/hist_NMNAT1.Score.png\")\n",
        "hist(UMAP_int$NMNAT1_ModuleScore1)\n",
        "dev.off()\n",
        "\n",
        "UMAP_int$NMNAT1_ModuleScore1_binary <- ifelse(UMAP_int$NMNAT1_ModuleScore1 > 0.06, 100, 0)\n",
        "\n",
        "\n",
        "pdf(\"~/Downloads/NMNAT1_ModuleScore1_BINARY_COLORSCALE.pdf\", width=5, height=5)\n",
        "FeaturePlot(UMAP_int, features=c(\"NMNAT1_ModuleScore1_binary\"),cols = c(c(\"orange\", \"red\")))\n",
        "dev.off()\n",
        "\n",
        "\n",
        "pdf(\"~/Downloads/NMNAT1.BINARY.dmso.tmz.pdf\", width=9, height=5)\n",
        "FeaturePlot(UMAP_int, features=c(\"NMNAT1_ModuleScore1_binary\"),cols = c(\"orange\", \"red\"), split.by=\"conds\")\n",
        "dev.off()\n"
      ],
      "metadata": {
        "id": "Jvlx4Y4oBKxT"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}